Code/Framework/AzFramework/AzFramework/Components/NativeUISystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+            AZ::Interface<InputDeviceMotion::ImplementationFactory>::Register(m_deviceMotionImplFactory);
+        }
+
+        if ((m_deviceMouseImplFactory = GetDeviceMouseImplentationFactory()) != nullptr)
+        {
+            AZ::Interface<InputDeviceMouse::ImplementationFactory>::Register(m_deviceMouseImplFactory);
+        }
+
+        if ((m_deviceTouchImplFactory = GetDeviceTouchImplentationFactory()) != nullptr)
+        {
+            AZ::Interface<InputDeviceTouch::ImplementationFactory>::Register(m_deviceTouchImplFactory);
+        }
+
+        if ((m_deviceVirtualKeyboardImplFactory = GetDeviceVirtualKeyboardImplentationFactory()) != nullptr)
+        {
+            AZ::Interface<InputDeviceVirtualKeyboard::ImplementationFactory>::Register(m_deviceVirtualKeyboardImplFactory);
+        }
+
+        if ((m_nativeWindowImplFactory = GetNativeWindowImplementationFactory()) != nullptr)
+        {
+            AZ::Interface<NativeWindow::ImplementationFactory>::Register(m_nativeWindowImplFactory);
+        }
+    }
+
+    NativeUISystemComponent::~NativeUISystemComponent()
+    {
+        if (m_applicationImplFactory != nullptr)
+        {
+            AZ::Interface<Application::ImplementationFactory>::Unregister(m_applicationImplFactory);
+            delete m_applicationImplFactory;
+        }
+
+        if (m_deviceGamepadImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceGamepad::ImplementationFactory>::Unregister(m_deviceGamepadImplFactory);
+            delete m_deviceGamepadImplFactory;
+        }
+        
+        if (m_deviceKeyboardImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceKeyboard::ImplementationFactory>::Unregister(m_deviceKeyboardImplFactory);
+            delete m_deviceKeyboardImplFactory;
+        }
+
+        if (m_deviceMotionImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceMotion::ImplementationFactory>::Unregister(m_deviceMotionImplFactory);
+            delete m_deviceMotionImplFactory;
+        }
+
+        if (m_deviceMouseImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceMouse::ImplementationFactory>::Unregister(m_deviceMouseImplFactory);
+            delete m_deviceMouseImplFactory;
+        }
+
+        if (m_deviceTouchImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceTouch::ImplementationFactory>::Unregister(m_deviceTouchImplFactory);
+            delete m_deviceTouchImplFactory;
+        }
+
+        if (m_deviceVirtualKeyboardImplFactory != nullptr)
+        {
+            AZ::Interface<InputDeviceVirtualKeyboard::ImplementationFactory>::Unregister(m_deviceVirtualKeyboardImplFactory);
+            delete m_deviceVirtualKeyboardImplFactory;
+        }
+
+        if (m_nativeWindowImplFactory != nullptr)
+        {
+            AZ::Interface<NativeWindow::ImplementationFactory>::Unregister(m_nativeWindowImplFactory);
+            delete m_deviceVirtualKeyboardImplFactory;
+        }
```

Comment:
repeat: here the if checks aren't needed.
If a nullptr is passed to [AZ::Interface<T>::Unregister](https://github.com/o3de/o3de/blob/ea1c05cd8a0e4916881bdf9cbbc66d0bc7188499/Code/Framework/AzCore/AzCore/Interface/Interface.h#L162-L163) it will just return a failure outcome string, which can then just be ignored.
```suggestion
        AZ::Interface<Application::ImplementationFactory>::Unregister(m_applicationImplFactory);
        m_applicationImplFactory.reset();

        AZ::Interface<InputDeviceGamepad::ImplementationFactory>::Unregister(m_deviceGamepadImplFactory);
        m_deviceGamepadImplFactory.reset();
        
        AZ::Interface<InputDeviceKeyboard::ImplementationFactory>::Unregister(m_deviceKeyboardImplFactory);
        m_deviceKeyboardImplFactory.reset();

        AZ::Interface<InputDeviceMotion::ImplementationFactory>::Unregister(m_deviceMotionImplFactory);
        m_deviceMotionImplFactory.reset();

        AZ::Interface<InputDeviceMouse::ImplementationFactory>::Unregister(m_deviceMouseImplFactory);
        m_deviceMouseImplFactory.reset();

        AZ::Interface<InputDeviceTouch::ImplementationFactory>::Unregister(m_deviceTouchImplFactory);
        m_deviceTouchImplFactory.reset();

        AZ::Interface<InputDeviceVirtualKeyboard::ImplementationFactory>::Unregister(m_deviceVirtualKeyboardImplFactory);
        m_deviceVirtualKeyboardImplFactory.reset();

        AZ::Interface<NativeWindow::ImplementationFactory>::Unregister(m_nativeWindowImplFactory);
        m_deviceVirtualKeyboardImplFactory.reset();
```